feat: Add expanded support for CSS Colors

gabrieldonadel opened the pull request:
## Summary 

This PR adds expanded support for CSS Colors, as requested on https://github.com/facebook/react-native/issues/34425. It updates the current regex used to match the functional notation for colors, e.g `rgb()` to accept space as a valid separator between values as specified on [CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/) definition. This also adds support for the `hwb` notation.

## Changelog


[General] [Added] - Add expanded support for CSS Colors

## Test Plan

We can test different color scenarios through the new test cases added to the already existing normalizeColor-test.js

necolas reviewed (commented):
We should add tests for `rgba(255 255 255 / 0.5)` syntax, and for space-separated `hsl`, `hsla`, etc.

cipolleschi reviewed (changes requested):
See comment from @necolas

gabrieldonadel commented:
> We should add tests for `rgba(255 255 255 / 0.5)` syntax, and for space-separated `hsl`, `hsla`, etc.

Sure, I've added a new regex so we can support the `/` notation and added new test cases
